### PR TITLE
Provide an option to limit the maximum concurrent request to consul.

### LINKED ISSUE #102: Error: too many open files
The number of concurrent queries to Consul is based on number of services.
This can easily exceed the maximum number of file descriptors, so sockets fail to open and the exporter fails to export all metrics.

Please can this be fixed by merging #94 which provides an option to limit concurrency?

### PR DESCRIPTION
**Pull request by MaruHyl**
@brian-brazil  If a large number of services are registered on the consul, it may cause too much load on the consul at once.

Closes #102

### CODE REVIEW

File: consul_exporter.go [hunk 1]
```diff
@@ -360,6 +375,7 @@ func main() {
 	kingpin.Flag("consul.key-file", "File path to a PEM-encoded private key used with the certificate to verify the exporter's authenticity.").Default("").StringVar(&opts.keyFile)
 	kingpin.Flag("consul.server-name", "When provided, this overrides the hostname for the TLS certificate. It can be used to ensure that the certificate name matches the hostname we declare.").Default("").StringVar(&opts.serverName)
 	kingpin.Flag("consul.timeout", "Timeout on HTTP requests to consul.").Default("200ms").DurationVar(&opts.timeout)
+	kingpin.Flag("consul.request-limit", "Limit the maximum number of concurrent requests to consul.").Default("0").IntVar(&opts.requestLimit)
```
Comment (grobie) [inline consul_exporter.go]: Do you have a suggestion for a default which prevents overloads out-of-the-box? I'd not leave it it unbounded by default.

Comment (MaruHyl) [reply consul_exporter.go]: I think it's better to set the default value to 1. This way the default load is minimal for the consul. 
    If the user wants to collect metrics faster, they can increase this value according to their environment.
    In my test environment , about one hundred services, and the request-limit is 1, the exporter can still complete the metrics collection within 200ms.

Comment (MaruHyl) [reply consul_exporter.go]: @grobie

**Comment by MaruHyl** [comment]
@grobie What do I need to do now？

**Comment by robincw-gr** [comment]
Fixes #102

**Comment by MaruHyl** [comment]
@robincw-gr ok, I will recommit later.

**Comment by brmurphy** [comment]
This looks great! Lets get it merged to master @grobie.

**Comment by simonpasquier** [comment]
@MaruHyl it seems that you've introduced unrelated commits in your PR. Are you willing to clean this up? Thanks!

**Comment by TopeltR** [comment]
@MaruHyl Hey! Will this feature still be considered to add in the near future? We experienced the same issue on k8s nodes, when we migrated the exporter from Swarm to K8s.

**Comment by MaruHyl** [comment]
@TopeltR ok

**Comment by roidelapluie** [comment]
dupe of #164

**Pull request closed without merging**